res/res_pjsip_rfc3326.c

```diff
@@ -52,24 +52,36 @@ static void rfc3326_use_reason_header(struct ast_sip_session *session, struct pj
 
 		cause_q850 = !strncasecmp(cause, "Q.850", 5);
 		cause_sip = !strncasecmp(cause, "SIP", 3);
+
 		if ((cause_q850 || cause_sip) && (cause = strstr(cause, "cause="))) {
-			/* If text is present get rid of it */
+			/* If text is present, process and get rid of it */
 			if ((text = strchr(cause, ';'))) {
+
+				if(!strncasecmp(text, "text=\"Call completed elsewhere\"", 31)) {
```

Comment:
Space between if and (, and why is this a strncasecmp of a specific length (add comment why)?

Reply:
strncasecmp accepts 3 arguments (the last one is the length), you can't just remove the argument.

What Josh is asking is where the number 31 came from. We don't like magic numbers just sprinkled in the code. Here, it's the length of the string so I guess to me it's obvious, but it may not be for everyone. A comment indicating what 31 means would help.

I'm not sure if gcc is smart enough to optimize away a call like strlen("text=\"Call completed elsewhere\"") in lieu of the length. I have a special macro for this that I use, that avoids having to hardcode numbers at all.

Reply:
The Q.850 and SIP checks do it so they don't have to copy and parse the string. In the case of answered elsewhere that shouldn't occur because the rest of the string should strictly be the string being checked. That's why I asked, to see if you had a case where the remaining part of the string was not:

`text=”Call completed elsewhere”`

As that could indicate other problems.

Reply:
What does that mean that it could be something different in reference to this change?

Reply:
If you are referring to a DIFFERENT text string for some other thing, then yes, of course it could be. I was strictly referring to the Call completed elsewhere case.

At this point just leave the strncasecmp.